### PR TITLE
UI cleanup

### PR DESCRIPTION
**Pull request by Kiisu-Master**
I described the changes in the commit message. 
Sorry for the one big commit, its probably hard to understand what changed.

### CODE REVIEW

**Comment by MewPurPur** [comment]
Please don't change the scaling mode yet.

File: src/GlobalSettings.gd [hunk 1]
```diff
@@ -73,6 +73,7 @@ func _enter_tree() -> void:
 	load_settings()
 	load_user_data()
 	DisplayServer.window_set_mode(save_data.window_mode)
+	get_window().wrap_controls = true
```
Comment (Kiisu-Master) [inline src/GlobalSettings.gd:76]: This enables automatic window minimum size.
I didnt know where else to put this

File: src/ui_parts/display.tscn [hunk 2]
```diff
@@ -110,31 +116,36 @@ allow_lower = false
 unique_name_in_owner = true
 layout_mode = 2
 
-[node name="ViewportContainer" type="SubViewportContainer" parent="."]
+[node name="ViewportPanel" type="PanelContainer" parent="."]
+layout_mode = 2
+size_flags_vertical = 3
+theme_override_styles/panel = SubResource("StyleBoxFlat_jk4ec")
+
+[node name="ViewportContainer" type="SubViewportContainer" parent="ViewportPanel"]
```
Comment (Kiisu-Master) [inline src/ui_parts/display.tscn:124]: A panel container to make the outline around viewport

File: src/ui_parts/main_scene.tscn [hunk 3]
```diff
@@ -51,3 +48,5 @@ layout_mode = 2
 [node name="Display" parent="HSplitContainer" instance=ExtResource("3_qbqbs")]
 unique_name_in_owner = true
 layout_mode = 2
+size_flags_horizontal = 3
+size_flags_stretch_ratio = 1.5
```
Comment (Kiisu-Master) [inline src/ui_parts/main_scene.tscn:52]: i removed split offset and made it use this stretch ratio instead, i hope its fine

File: src/GlobalSettings.gd (hunk 1 above)
Comment (MewPurPur) [reply src/GlobalSettings.gd:76]: What does this mean?

Comment (Kiisu-Master) [reply src/GlobalSettings.gd:76]: you cant resize the window to something too small to fit the content anymore

File: src/ui_parts/main_scene.tscn (hunk 3 above)
Comment (MewPurPur) [reply src/ui_parts/main_scene.tscn:52]: I'd like for the viewport to be exactly square by default, does this affect that?

Comment (Kiisu-Master) [reply src/ui_parts/main_scene.tscn:52]: oh well, probably 💀

File: src/GlobalSettings.gd (hunk 1 above)
Comment (MewPurPur) [reply src/GlobalSettings.gd:76]: This is cool, let's keep it, just add a comment.

Comment (Kiisu-Master) [reply src/GlobalSettings.gd:76]: its only going to be useful with the scaling mode change tho.
ill add a comment right now

Comment (MewPurPur) [reply src/GlobalSettings.gd]: Just tested it, no, it's useful without the scaling mode change. Also, I still want to change the scaling mode, I just want to not do it without implementing customizable theme scale.

**Comment by MewPurPur** [comment]
Alrighty, I can merge this after the above concern is addressed and it's rebased.

**Comment by MewPurPur** [comment]
Can this be merged now?

**Comment by MewPurPur** [comment]
YOLO

**Pull request merged**